src/app.js

```diff
@@ -222,15 +237,35 @@ class SAFEApp extends EventEmitter {
           file = await emulation.fetch(filePath);
         }
         const openedFile = await emulation.open(file, consts.pubConsts.NFS_FILE_MODE_READ);
-        const data = await openedFile.read(
-          consts.pubConsts.NFS_FILE_START, consts.pubConsts.NFS_FILE_END);
+        const fileSize = await openedFile.size() || '*';
+        let data;
+
+        // TODO: how do we handle multipart Reqs
```

Comment:
We can either implement it here and have the `range` object to be renamed to `ranges` and be an array of objects instead, or just don't implement such a support here but in Peruse/browser, the former would be more efficient though as we would perform the steps of finding the root container for the service only once.

Reply:
I sent a PR (https://github.com/joshuef/safe_app_nodejs/pull/2) against your branch @joshuef which backs up my comments plus some more tests I created. I found that safe_app lib is not validating the start offset so we'll need to raise that as an issue, for now I skipped the tests corresponding to those scenarios.
You will see that I made an adjustment to the values returned in the `Content-Range` header as per my understanding of it.